src/Operator.ts

```diff
@@ -174,30 +175,12 @@ export default class Operator {
   /**
    * Add two operators.
    *
-   * Note: May be overengineered for adding 2 vectors with this map-reduce approach.
-   *
    * @param m2 Other operator with same dimensions.
    * @returns m1 + m2
+   * @see {@link Operator.add} for the actual implementation.
    */
   add(m2: Operator): Operator {
-    const m1 = this
-
-    Dimension.checkDimensions(m1.dimensionsIn, m2.dimensionsIn)
-    Dimension.checkDimensions(m1.dimensionsOut, m2.dimensionsOut)
-
-    const entries = _.chain(m1.entries.concat(m2.entries))
-      .groupBy((entry: OperatorEntry) => `${entry.coordOut.toString()}-${entry.coordIn.toString()} `)
-      .values()
-      .map((grouped: OperatorEntry[]) => {
-        const coordOut = [...grouped[0].coordOut]
-        const coordIn = [...grouped[0].coordIn]
-        const value = grouped.map((entry) => entry.value).reduce((a, b) => a.add(b))
-        return new OperatorEntry(coordOut, coordIn, value)
-      })
-      .filter((entry) => !entry.value.isAlmostZero())
-      .value()
-
-    return new Operator(entries, m1.dimensionsOut, m1.dimensionsIn)
+    return Operator.add([this, m2])
```

Comment:
I understand that performance is crucial.
However, I think if it makes sense to preserve a more readable implementation (even if only as a reference for tests).

Reply:
I don't want to keep two versions of the same method around, especially if one will not be used. It's going to be a maintenance burden. If you don't find the existing one readable enough, I should maybe work towards making it cleaner or possible add a few comments.

Don't get me wrong, I'm a fan of functional style iterators and transformers, but those are just awfuly slow in JS. The compiler simply doesn't have enough time and information to optimize those properly. If performance of some code doesn't matter in practice, then by all means use them.

Reply:
Yes, optimization and readability are often at odds.

My experience with algorithms & numerics is that it makes sense to have simpler/easier implementations, even if these are much slower (e.g. `n^2` not `n log(n)`). First, it works for testing. Second, it makes it simpler to expand (or optimize) code in a different way. Those methods are not intended to be used outside in regular applications (a clear note is required).

Compare with `propagatePhotons()` for the fast (and actually used) method, and a different one for checking it actually works correctly (https://github.com/Quantum-Game/quantum-tensors/blob/master/src/Photons.ts#L225-L240).

Reply:
I do get it for `propagatePhotons`, as those algorithms are radically different and the logic is pretty complex. I don't believe that's the case for operations like dot and add. The operations are pretty simple and tests do cover them extensively enough as a reference. On the other hand, if we ever change the storage implementation for vectors, maintaining two versions of those will be a pain.

Also I have to note that the previous logic was no easier for me to understand. Both implementations in my eyes would be likely to get wrong easily, which means neither is really a better reference for the other. If that wasn't JS, we could certainly build/use a better primitive to implement those much more clearly without the performance regression.